src/wallet3/wallet2½.cpp

```diff
@@ -0,0 +1,52 @@
+#include "wallet2½.hpp"
+
+#include <ringct/rctSigs.h>
+#include <cryptonote_basic/cryptonote_format_utils.h>
+
+namespace wallet2½
+{
+  // TODO: copied from wallet2 for now, so to not redo crypto stuff just yet.
+  //
+  // Definitely quite nasty and needs revisited.
+  uint64_t
+  output_amount(
+      const rct::rctSig& rv,
+      const crypto::key_derivation& derivation,
+      unsigned int i,
+      rct::key& mask,
+      hw::device& hwdev)
+  {
+    crypto::secret_key scalar1;
+    hwdev.derivation_to_scalar(derivation, i, scalar1);
+    switch (rv.type)
+    {
+      case rct::RCTType::Simple:
+      case rct::RCTType::Bulletproof:
+      case rct::RCTType::Bulletproof2:
+      case rct::RCTType::CLSAG:
+        return rct::decodeRctSimple(rv, rct::sk2rct(scalar1), i, mask, hwdev);
+      case rct::RCTType::Full:
+        return rct::decodeRct(rv, rct::sk2rct(scalar1), i, mask, hwdev);
+      default:
+        throw std::invalid_argument("Unsupported rct type");
+    }
+  }
+
+  crypto::hash
+  tx_hash(const cryptonote::transaction& tx)
+  {
+    crypto::hash h;
```

Comment:
value initialize

Reply:
Or better yet completely avoid the local variable with:
```C++
return cryptonote::get_transaction_hash(tx);
```
(though that makes me wonder -- is this function just aiming to provide a shortcut that already exists?)